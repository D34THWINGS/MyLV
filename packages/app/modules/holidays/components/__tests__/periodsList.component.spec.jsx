import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import { PeriodsList } from '../periodsList.component'

jest.unmock('../periodsList.component')

describe('PeriodsList', () => {
  let props
  beforeEach(() => {
    props = {
      classes: {
        periodsList: 'periods-list',
      },
      fields: {
        get: jest.fn(index => ({ _id: index })),
        push: jest.fn(),
        remove: jest.fn(),
        map: jest.fn(fn => [fn({
          label: 'foo',
          startDate: 'bar',
          endDate: 'baz',
          _id: 123,
        }, 0)]),
      },
    }
  })

  const getWrapper = () => shallow(<PeriodsList {...props} />)

  it('should render properly', () => {
    // When
    const wrapper = getWrapper()

    // Then
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
