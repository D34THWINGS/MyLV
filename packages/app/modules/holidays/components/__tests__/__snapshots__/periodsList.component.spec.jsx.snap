// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeriodsList should render properly 1`] = `
<WithStyles(Paper)
  className="periods-list"
>
  <div>
    <WithStyles(Table)>
      <WithStyles(TableHead)>
        <WithStyles(TableRow)>
          <WithStyles(TableCell)
            key="paidHolidays"
            padding="checkbox"
          >
            Congés payés
          </WithStyles(TableCell)>
          <WithStyles(TableCell)
            key="unpaidHolidays"
            padding="checkbox"
          >
            Congés sans solde
          </WithStyles(TableCell)>
          <WithStyles(TableCell)
            key="conventionalHolidays"
            padding="checkbox"
          >
            Absences Syntec
          </WithStyles(TableCell)>
          <WithStyles(TableCell)
            padding="checkbox"
          >
            Date du premier jour d'absence
          </WithStyles(TableCell)>
          <WithStyles(TableCell)>
            A partir du
          </WithStyles(TableCell)>
          <WithStyles(TableCell)
            padding="checkbox"
          >
            Date du dernier jour d'absence
          </WithStyles(TableCell)>
          <WithStyles(TableCell)>
            Jusqu'au
          </WithStyles(TableCell)>
          <WithStyles(TableCell)
            padding="checkbox"
          />
        </WithStyles(TableRow)>
      </WithStyles(TableHead)>
      <WithStyles(TableBody)>
        <WithStyles(TableRow)
          key="0"
        >
          <WithStyles(TableCell)
            key="paidHolidays"
            padding="checkbox"
          >
            <Field
              component={[Function]}
              name="[object Object].label"
              type="radio"
              value="paidHolidays"
            />
          </WithStyles(TableCell)>
          <WithStyles(TableCell)
            key="unpaidHolidays"
            padding="checkbox"
          >
            <Field
              component={[Function]}
              name="[object Object].label"
              type="radio"
              value="unpaidHolidays"
            />
          </WithStyles(TableCell)>
          <WithStyles(TableCell)
            key="conventionalHolidays"
            padding="checkbox"
          >
            <Field
              component={[Function]}
              name="[object Object].label"
              type="radio"
              value="conventionalHolidays"
            />
          </WithStyles(TableCell)>
          <WithStyles(TableCell)
            padding="checkbox"
          >
            <Field
              component={[MockFunction]}
              name="[object Object].startDate"
              outlined={false}
              type="date"
            />
          </WithStyles(TableCell)>
          <WithStyles(TableCell)>
            <div>
              <span>
                Matin
              </span>
              <Field
                component={[Function]}
                name="[object Object].startOnPM"
                type="checkbox"
                value="startOnPM"
              />
              <span>
                Midi
              </span>
            </div>
          </WithStyles(TableCell)>
          <WithStyles(TableCell)
            padding="checkbox"
          >
            <Field
              component={[MockFunction]}
              name="[object Object].endDate"
              outlined={false}
              type="date"
            />
          </WithStyles(TableCell)>
          <WithStyles(TableCell)>
            <div>
              <span>
                Midi
              </span>
              <Field
                component={[Function]}
                name="[object Object].endOnPM"
                type="checkbox"
                value="endOnPM"
              />
              <span>
                Soir
              </span>
            </div>
          </WithStyles(TableCell)>
          <WithStyles(TableCell)
            padding="checkbox"
          />
        </WithStyles(TableRow)>
      </WithStyles(TableBody)>
    </WithStyles(Table)>
  </div>
  <table>
    <WithStyles(TableFooter)>
      <WithStyles(TableRow)>
        <WithStyles(TableCell)>
          <WithStyles(Button)
            color="primary"
            onClick={[Function]}
          >
            Ajouter une ligne
          </WithStyles(Button)>
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
    </WithStyles(TableFooter)>
  </table>
</WithStyles(Paper)>
`;
